### PR TITLE
GOOWOO-706: Move default times prefill to `SavedSetupStepper`

### PR DESCRIPTION
**Pull request by asvinb**
### Changes proposed in this Pull Request:

Closes https://linear.app/a8c/issue/GOOWOO-706/estimated-shipping-times-are-blank

_Replace this with a good description of your changes & reasoning._

### Detailed test instructions:

1. Go to the onboarding
2. In the second step, without changing the default shipping times, click on continue
3. There should not be any validation error on the estimate shipping times as they are prefilled with data
4. If no countries are selected, the shipping rates and times section should not show up.

### Changelog entry

> Fix - Estimated shipping times no longer show validation errors when continuing onboarding without changing the prefilled defaults.

### CODE REVIEW

File: js/src/pages/onboarding/setup-stepper/saved-setup-stepper.js [hunk 1]
```diff
@@ -88,6 +92,35 @@ const SavedSetupStepper = ( { savedStep } ) => {
...
+			targetAudience?.location
+		) {
+			const countries = getFinalCountriesRef.current( targetAudience );
+			if ( countries.length ) {
+				const defaultTimes = countries.map( ( countryCode ) => ( {
+					countryCode,
+					time: DEFAULT_SHIPPING_MIN_TIME,
+					maxTime: DEFAULT_SHIPPING_MAX_TIME,
+				} ) );
+				saveShippingTimes( defaultTimes );
```
Comment (mserino) [inline js/src/pages/onboarding/setup-stepper/saved-setup-stepper.js]: Does `saveShippingTimes` catch and handle errors? Should we use a `try catch` block and `createNotice` in case of error?

Comment (asvinb) [reply js/src/pages/onboarding/setup-stepper/saved-setup-stepper.js]: Good catch @mserino . Updated!

**Review by mserino — APPROVED** [review]
Please also see my other comments here: https://github.com/woocommerce/google-listings-and-ads/pull/3524#pullrequestreview-4562721557

**Comment by asvinb** [comment]
@mserino Let's get this one merged first after successful QA and then we can merge back in the multilingual branch 👍

**Comment by ankitguptaindia** [comment]
QA Status: Needs Feedback ⚠️ 
----

Thanks for the PR, @asvinb! It fixes the original validation issue.

However, I noticed some cases when a merchant removes a country and then re-adds it:

1. **Estimated shipping times**

   * When the country is removed, the **Estimated shipping times** section is hidden as expected.
   * However, after re-adding the country, the section initially displays a validation error without the day fields. A moment later, the day fields are automatically added.

   Screenshot 2026-06-26 at 8 50 43 PM

2. **Estimated shipping rates**

   * After re-adding the country, the **Estimated shipping rates** field displays the following validation error:

     > `Please specify estimated shipping rates for all the countries, and the rate cannot be less than 0.`
   * If the user simply clicks inside the text box, the field is automatically populated and the validation error disappears without any user input.

Could you take a look at these edge cases as well?



### Test Steps in screencast:  




https://github.com/user-attachments/assets/02393f12-1300-4db0-9ea5-baeb61a6fd13

**Comment by ankitguptaindia** [comment]
### QA Status: Verified ✅ 


**QA Summary:**

* ✅ Verified that clicking **Continue** on the shipping step without modifying the default **Estimated shipping times** no longer triggers any validation errors, as the fields are prefilled.
* ✅ Verified that after removing and re-adding a country, the **Estimated shipping times** fields works correctly without displaying any temporary validation errors.
* ✅ Verified that after re-adding a country, the **Estimated shipping rates** field works correctly and no validation error is displayed.
* ✅ Confirmed the onboarding flow works as expected across the tested scenarios.


- Testing performed on `feature/multi-lingual-support`


https://github.com/user-attachments/assets/761b4a5a-6eff-4e56-b3bc-5857b21430a4

**Review by ankitguptaindia — APPROVED** [review]
